Utility/Mocks.lua

```diff
@@ -0,0 +1,69 @@
+--- Stubs out a mocked method.
```

Comment:
This file contains methods used to configure the mocks.

Basically it does some magic to create functions on the mock that will allow you to call into the fake API as well as change the values that the fake API returns.